lens/private/syntax/syntax.rkt

```diff
@@ -53,14 +53,14 @@
       [(a ...) (ormap (target-stx target-id) (syntax->list #'(a ...)))]
       [a (and (bound-identifier=? target-id #'a) #'a)]))
   
-  (define template->pattern
-    (syntax-parser #:literals (_)
-      [(a ...) #`(#,@(stx-map template->pattern #'(a ...)))]
-      [_ (generate-temporary)]
+  (define (template->pattern stx)
+    (syntax-parse stx
+      [(a ...) (datum->syntax stx (stx-map template->pattern #'(a ...)) stx stx)]
```

Comment:
Oh. This `datum->syntax` change isn't strictly necessary; I just thought it would be good to change it to keep as much of the lexical context and properties as possible. 

The change on the next line to use `(~literal _)` is necessary because of how `syntax-parse` used to work, but I think it's more explicit now and that's a good thing.